File: LeetCode/Easy/1394-Find_Lucky_Integer_In_An_Array/1394-FindLuckyIntegerInAnArray.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var findLucky = function (arr) {
    let freq = new Map();

    for (let num of arr)
        freq.set(num, (freq.get(num) || 0) + 1);

    let ans = -1;

    for (let [k, v] of freq) {
        if (k == v && k > ans)
            ans = k;
    }

    return ans;
};

File: LeetCode/Easy/1394-Find_Lucky_Integer_In_An_Array/1394-FindLuckyIntegerInAnArray.py
from collections import Counter


class Solution:
    def findLucky(self, arr: list[int]) -> int:
        freq = Counter(arr)

        ans = -1

        for k, v in freq.items():
            if k == v and k > ans:
                ans = k

        return ans


a = Solution()
print(a.findLucky(arr=[2, 2, 3, 4]))
print(a.findLucky(arr=[1, 2, 2, 3, 3, 3]))
print(a.findLucky(arr=[2, 2, 2, 3, 3]))
